frontend/src/pages/instance/Snapshots/index.vue

```diff
@@ -259,13 +256,15 @@ export default {
             this.$emit('instance-updated')
         },
         async downloadSnapshotPackage (snapshot) {
-            const ss = await SnapshotApi.getSnapshot(this.instance.id, snapshot.id)
+            const ss = await SnapshotsApi.getFullSnapshot(snapshot.id)
+            const owner = snapshot.device || snapshot.project
+            const ownerType = snapshot.device ? 'device' : 'instance'
             const packageJSON = {
-                name: this.instance.safeName,
-                description: `${snapshot.name} - ${snapshot.description}`,
+                name: `${owner.safeName || owner.name}`.replace(/[^a-zA-Z0-9-]/g, '-').toLowerCase(),
+                description: `${ownerType} snapshot, ${snapshot.name} - ${snapshot.description}`,
                 private: true,
                 version: '0.0.0-' + snapshot.id,
-                dependencies: ss.modules
+                dependencies: ss.settings?.modules || {}
             }
             const element = document.createElement('a')
```

Comment:
We could use the same download snapshot composable trigger

Reply:
resolved in [15001ab](https://github.com/FlowFuse/flowfuse/pull/3885/commits/15001ab22c59d4581f2116b97633c5c418756099)